linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -29,33 +29,41 @@ import org.scalajs.ir.Trees.{MemberNamespace, JSNativeLoadSpec}
 import org.scalajs.ir.Types.ClassRef
 
 import org.scalajs.linker._
+import org.scalajs.linker.frontend.IRLoader
 import org.scalajs.linker.interface.{ESVersion, ModuleKind, ModuleInitializer}
 import org.scalajs.linker.interface.unstable.ModuleInitializerImpl
 import org.scalajs.linker.standard._
 import org.scalajs.linker.standard.ModuleSet.ModuleID
 
+import org.scalajs.logging.Logger
+
 import Analysis._
 import Infos.{NamespacedMethodName, ReachabilityInfo, ReachabilityInfoInClass}
 
-private final class Analyzer(config: CommonPhaseConfig,
-    moduleInitializers: Seq[ModuleInitializer],
-    symbolRequirements: SymbolRequirement,
-    allowAddingSyntheticMethods: Boolean,
-    checkAbstractReachability: Boolean,
-    infoLoader: InfoLoader,
-    ec: ExecutionContext)
-    extends Analysis {
+final class Analyzer(config: CommonPhaseConfig, initial: Boolean,
+    checkIR: Boolean, irLoader: IRLoader) extends Analysis {
```

Comment:
We should hide the `extends Analysis` from clients of `Analyzer`, now that the class is actually intended to be used elsewhere. It could be as simple as declaring an inner `private object AnalysisImpl extends Analysis` that forwards its methods to the enclosing `Analyzer`.